src/client/api/init.c

```diff
@@ -93,19 +102,27 @@ const struct daos_task_api dc_funcs[] = {
 	{dc_obj_punch_akeys, sizeof(daos_obj_punch_t)},
 	{dc_obj_query, sizeof(daos_obj_query_t)},
 	{dc_obj_query_key, sizeof(daos_obj_query_key_t)},
+	{dc_obj_sync, sizeof(struct daos_obj_sync)},
 	{dc_obj_fetch_shard, sizeof(struct daos_obj_fetch_shard)},
 	{dc_obj_fetch, sizeof(daos_obj_fetch_t)},
 	{dc_obj_update, sizeof(daos_obj_update_t)},
 	{dc_obj_list_dkey, sizeof(daos_obj_list_dkey_t)},
 	{dc_obj_list_akey, sizeof(daos_obj_list_akey_t)},
 	{dc_obj_list_rec, sizeof(daos_obj_list_recx_t)},
+	{dc_obj_list_obj, sizeof(daos_obj_list_obj_t)},
+
+	/** Array */
 	{dc_array_create, sizeof(daos_array_create_t)},
 	{dc_array_open, sizeof(daos_array_open_t)},
 	{dc_array_close, sizeof(daos_array_close_t)},
+	{dc_array_destroy, sizeof(daos_array_destroy_t)},
 	{dc_array_read, sizeof(daos_array_io_t)},
 	{dc_array_write, sizeof(daos_array_io_t)},
+	{dc_array_punch, sizeof(daos_array_io_t)},
 	{dc_array_get_size, sizeof(daos_array_get_size_t)},
 	{dc_array_set_size, sizeof(daos_array_set_size_t)},
+
+	/** HL */
```

Comment:
It is not related with the patch, but we missed related logic, and suspect that it may cause some strange test behavior, so I add them into the patch.